flang/lib/Parser/prescan.cpp

```diff
@@ -1289,29 +1289,49 @@ const char *Prescanner::FreeFormContinuationLine(bool ampersand) {
     return nullptr;
   }
   p = SkipWhiteSpace(p);
-  if (InCompilerDirective()) {
-    if (*p++ != '!') {
-      return nullptr;
-    }
-    for (const char *s{directiveSentinel_}; *s != '\0'; ++p, ++s) {
-      if (*s != ToLowerCaseLetter(*p)) {
+  if (*p == '!') {
+    if (InCompilerDirective()) {
+      if (*p++ != '!') {
         return nullptr;
       }
```

Comment:
```suggestion
    ++p;
    if (InCompilerDirective()) {
```
We already know that `*p` is `'!'`.

Reply:
Thank you for the review.
I have implemented all the suggested changes.